### PR TITLE
Optional BLE interface to communicate with Espruino

### PR DESCRIPTION
**Pull request by mjdietzx**
This implements both a BLE Interface and the standard Serial interface for the nRF52832 Preview Development kit.

Now the user can choose to communicate with Espruino over the air through Bluetooth low energy. This means user can send commands like digitalWrite(18, 1);

to the nRF52 chip with their smartphone (or computer) instead of the Espruino IDE, no wires required.

Recommended to use nRF Toolbox app https://play.google.com/store/apps/details?id=no.nordicsemi.android.nrftoolbox&hl=en and specifically the UART module. Note softdevice must be programmed onto nRF52 before Espruino is programmed but i am working on fixing this.

Also not hex files are programmed to nRF52 using nRFgo Studio https://www.nordicsemi.com/eng/nordic/Products/nRFgo-Studio/nRFgo-Studio-Win64/14964.

Also i will improve instructions.

No features implemented in jshardware.c yet, just communication and write to GPIO.

### CODE REVIEW

**Comment by gfwilliams** [comment]
Wow, that's awesome! Looks like I'm going to have to get an NRF52 devkit now :) Just noticed some minor things though - I'll use the inline comment thing to add them,

File: Makefile [hunk 1]
```diff
@@ -68,7 +68,7 @@ endif
 INCLUDE=-I$(ROOT) -I$(ROOT)/targets -I$(ROOT)/src -I$(ROOT)/gen
 LIBS=
 DEFINES=
-CFLAGS=-Wall -Wextra -Wconversion -Werror=implicit-function-declaration
+CFLAGS=#-Wall -Wextra -Wconversion -Werror=implicit-function-declaration
```
Comment (gfwilliams) [inline Makefile]: Could we keep this? or maybe just do `CFLAGS=` in the NRF52 ifdef lower down?

File: targetlibs/nrf52/nRF52_SDK_0.9.1_3639cc9/components/toolchain/gcc/linker_ble_interface_espruino.ld [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+SEARCH_DIR(.)
+GROUP(-lgcc -lc -lnosys)
+
+MEMORY
+{
+  FLASH (rx) : ORIGIN = 0x1f000, LENGTH = 0x61000
+  RAM (rwx) :  ORIGIN = 0x20002800, LENGTH = 0x5800
+}
+
+INCLUDE "/home/michael/Documents/Espruino/targetlibs/nrf52/nRF52_SDK_0.9.1_3639cc9/components/toolchain/gcc/linker_common_espruino.ld"
```
Comment (gfwilliams) [inline targetlibs/nrf52/nRF52_SDK_0.9.1_3639cc9/components/toolchain/gcc/linker_ble_interface_espruino.ld]: absolute path - probably not what you wanted?

File: targetlibs/nrf52/nRF52_SDK_0.9.1_3639cc9/components/toolchain/gcc/linker_espruino.ld [hunk 3]
```diff
@@ -9,177 +9,4 @@
...
-   * size of stack_dummy section */
-  __StackTop = ORIGIN(RAM) + LENGTH(RAM);
-  _estack = __StackTop;
-  __StackLimit = __StackTop - SIZEOF(.stack_dummy);
-  PROVIDE(__stack = __StackTop);
-  
-  /* Check if data + heap + stack exceeds RAM limit */
-  ASSERT(__StackLimit >= __HeapLimit, "region RAM overflowed with stack")
-}
+INCLUDE "/home/michael/Documents/Espruino/targetlibs/nrf52/nRF52_SDK_0.9.1_3639cc9/components/toolchain/gcc/linker_common_espruino.ld"
```
Comment (gfwilliams) [inline targetlibs/nrf52/nRF52_SDK_0.9.1_3639cc9/components/toolchain/gcc/linker_espruino.ld]: another absolute path...

**Comment by mjdietzx** [comment]
Hey, yeah I forgot to remove that. Ill fix those paths etc.. good catch.

**Comment by gfwilliams** [comment]
Great, thanks! Didn't see the commit until just now :)

**Pull request merged**